### PR TITLE
Indirect ILL: Background subtraction from vanadium run

### LINKED ISSUE #19448: Indirect ILL, Background subtraction from vanadium
Possibility to subtract background also from vanadium is requested for indirect ILL reductions.

### PR DESCRIPTION
**Pull request by gvardany**
[files.zip](https://github.com/mantidproject/mantid/files/1014766/files.zip)

**To test:**
1. Set ILL/IN16B as instrument
2. Go to Interfaces>Indirect>Data Reduction>Energy Transfer Tab
3. Choose from the uploaded files as follows and click Run:
  - Sample : 090661
  - Calibration : 090663
  - Calibration background : 090658
  - Scaling Factor : 0.1
  - Leave the rest to defaults
4. Output should contain a single QENS peak in each spectra

Fixes #19448 .

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] How do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant documentation been added/updated?
- [x] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DavidFair** [comment]
I've downloaded the attached zip however run 091515 is missing?

File: MantidQt/CustomInterfaces/src/Indirect/ILLEnergyTransfer.cpp [hunk 1]
```diff
@@ -91,6 +85,26 @@ bool ILLEnergyTransfer::validate() {
...
+  // Validate the calibration background file
+  if (!m_uiForm.rfBackCalibrationRun->isValid()) {
+    uiv.addErrorMessage("Background run for calibration is invalid.");
+  } else {
+    bool isDouble = true;
+    m_backCalibScaling = m_uiForm.leBackCalibScale->text().toDouble(&isDouble);
+    if ((!isDouble || m_backCalibScaling <= 0) &&
+        !m_uiForm.rfBackCalibrationRun->getUserInput().toString().isEmpty()) {
+      uiv.addErrorMessage(
+          "Scale factor for calibration background is invalid.");
```
Comment (DavidFair) [inline MantidQt/CustomInterfaces/src/Indirect/ILLEnergyTransfer.cpp]: Can we inform the user that the scale factor must be a number and > 0

File: Testing/SystemTests/tests/analysis/ILLIndirectReductionFWS.py [hunk 2]
```diff
@@ -114,3 +115,13 @@ def _run_efws_mirror_sense(self):
...
+
+    def _run_calib_bg(self):
+        # this tests with the background file for calibration
+        IndirectILLReductionFWS(Run="143720:143728:2",
+                                CalibrationRun="143721",
+                                CalibrationBackgroundRun="143723",
+                                CalibrationBackgroundScalingFactor=0.1,
+                                OutputWorkspace="efws_calib_bg")
+
+        self.assertEquals(mtd['efws_calib_bg_red'].getItem(0).getNumberHistograms(), 18)
```
Comment (DavidFair) [inline Testing/SystemTests/tests/analysis/ILLIndirectReductionFWS.py]: Can we assert some Y data values so future developers are sure their changes don't affect the results.

File: Testing/SystemTests/tests/analysis/ILLIndirectReductionQENS.py [hunk 3]
```diff
@@ -135,7 +135,19 @@ def runTest(self):
...
 
+    def runTestBackgroundCalibration(self):
+
+        IndirectILLReductionQENS(Run="136558",
+                                 CalibrationRun="136553",
+                                 CalibrationBackgroundRun="136599",
+                                 CalibrationBackgroundScalingFactor=0.1,
+                                 OutputWorkspace='out_calib_bg')
+
+        self.assertEquals(mtd['out_calib_bg_red'].getItem(0).getNumberHistograms(), 18)
```
Comment (DavidFair) [inline Testing/SystemTests/tests/analysis/ILLIndirectReductionQENS.py]: ^ Assert on Y Data values

**Review by DavidFair — CHANGES_REQUESTED** [review]

**Comment by gvardany** [comment]
I have corrected the run numbers.

**Comment by DavidFair** [comment]
Testing verifies that the background for the calibration is being applied correctly. I also checked the validation and that  also handles bad values correctly.
Its just the a few minor points picked up in the code review otherwise everything looks good.

**Comment by gvardany** [comment]
I have addressed the comments.

**Comment by DavidFair** [comment]
Everything looks good to me. Once the builds finish :shipit:

**Review by DavidFair — APPROVED** [review]

**Comment by AndreiSavici** [comment]
Unrelated failure on `SliceViewerPythonInterfaceTest`

**Pull request merged**